addons/product/models/product.py

```diff
@@ -465,9 +465,13 @@ def _name_get(d):
             for r in supplier_info:
                 supplier_info_by_template.setdefault(r.product_tmpl_id, []).append(r)
         for product in self.sudo():
-            variant = product.product_template_attribute_value_ids._get_combination_name()
+            name = product.name
+            variant = ""
+            product_template_variant_values = product.product_template_attribute_value_ids._without_no_variant_attributes()
+            if product_template_variant_values:
+                variant = ", ".join(product_template_variant_values.mapped('name'))
+                name = "%s (%s)" % (product.name, variant)
```

Comment:
Note that going from 247.619s to 148.291s to load barcode still does not fundamentally change the problem that reading all the products of the database in one go does not look like a good idea.

Though I agree that any performance again is welcome, especially of this level. Just would be ideal to do it in a way that doesn't change features if possible.

Just an idea because I didn't test, but about the number of queries specifically, can't you just prefetch "manually" before the loop everything that will be used in `_get_combination_name` and leave the rest intact?

Reply:
> Note that going from 247.619s to 148.291s to load barcode still does not fundamentally change the problem that reading all the products of the database in one go does not look like a good idea.

I agree. See https://github.com/odoo/enterprise/pull/16905#issuecomment-795234504

> Just would be ideal to do it in a way that doesn't change features if possible.

I agree. Idea of this PR was proposed by FP

> Just an idea because I didn't test, but about the number of queries specifically, can't you just prefetch "manually" before the loop everything that will be used in _get_combination_name and leave the rest intact?

As I understand, it may decrease query time only, while most time consuming is in python scripts.